src/datasources/email/email.datasource.spec.ts

```diff
@@ -187,4 +188,73 @@ describe('Email Datasource Tests', () => {
       }),
     ).rejects.toThrow(EmailAddressDoesNotExistError);
   });
+
+  it('gets only verified email addresses associated with a given safe address', async () => {
+    const chainId = faker.number.int({ max: DB_CHAIN_ID_MAX_VALUE }).toString();
+    const safeAddress = faker.finance.ethereumAddress();
+    const verifiedSigners = [
+      {
+        emailAddress: faker.internet.email(),
+        signer: faker.finance.ethereumAddress(),
+        code: faker.number.int({ max: 999998 }).toString(),
+      },
+      {
+        emailAddress: faker.internet.email(),
+        signer: faker.finance.ethereumAddress(),
+        code: faker.number.int({ max: 999998 }).toString(),
+      },
+    ];
+    const nonVerifiedSigners = [
+      {
+        emailAddress: faker.internet.email(),
+        signer: faker.finance.ethereumAddress(),
+        code: faker.number.int({ max: 999998 }).toString(),
+      },
+      {
+        emailAddress: faker.internet.email(),
+        signer: faker.finance.ethereumAddress(),
+        code: faker.number.int({ max: 999998 }).toString(),
+      },
+    ];
+    await Promise.all(
+      verifiedSigners.map(async ({ emailAddress, signer, code }) => {
+        await target.saveEmail({
```

Comment:
Here we are waiting for the `saveEmail` before moving to the next iteration. Therefore `Promise.all` does not seem to "help" much.

I would however not use `Promise.all` here since it doesn't seem to provide a lot of benefits. Order of insertion is lost if the scheduler switches between the different `saveEmail` right?

Reply:
Yes, maybe it is a bit overkill to iterate using a `Promise.await` and a `.map` here. 

I've changed it in [f1ff610](https://github.com/safe-global/safe-client-gateway/pull/877/commits/f1ff61004d4b054ac07ac7b5750cec600b9d5d8a)